src/Service.Tests/Authorization/GraphQL/CreateMutationAuthorizationTests.cs

```diff
@@ -0,0 +1,390 @@
+                isAuthenticated: true,
+                clientRoleHeader: "test_role_with_excluded_fields_on_create");
+        }
+
+        #endregion
+
+        #region Multiple create mutation tests
+        /// <summary>
+        /// Test to validate that a 'create one' mutation request can only execute successfully when the user, has create permission
+        /// for all the entities involved in the mutation.
+        /// </summary>
+        [TestMethod]
+        public async Task ValidateAuthZCheckOnEntitiesForCreateOneMultipleMutations()
+        {
+            string createBookMutationName = "createbook";
+            string createOneBookMutation = @"mutation {
+                    createbook(item: { title: ""Book #1"", publishers: { name: ""Publisher #1""}}) {
+                        id
+                        title
+                    }
+                }";
+
+            // The anonymous role has create permissions on the Book entity but not on the Publisher entity.
+            // Hence the request will fail during authorization check.
+            await ValidateRequestIsUnauthorized(
+                graphQLMutationName: createBookMutationName,
+                graphQLMutation: createOneBookMutation,
+                expectedExceptionMessage: "Unauthorized due to one or more fields in this mutation.",
+                expectedExceptionStatusCode: DataApiBuilderException.SubStatusCodes.AuthorizationCheckFailed.ToString(),
+                isAuthenticated: false,
+                clientRoleHeader: "anonymous"
+                );
+
+            // The authenticates role has create permissions on both the Book and Publisher entities.
+            // Hence the authorization checks will pass.
+            await ValidateRequestIsAuthorized(
+                graphQLMutationName: createBookMutationName,
+                graphQLMutation: createOneBookMutation,
+                isAuthenticated: true,
+                clientRoleHeader: "authenticated"
+                );
+        }
+
+        /// <summary>
+        /// Test to validate that a 'create multiple' mutation request can only execute successfully when the user, has create permission
+        /// for all the entities involved in the mutation.
+        /// </summary>
+        [TestMethod]
+        public async Task ValidateAuthZCheckOnEntitiesForCreateMultipleMutations()
+        {
+            string createMultipleBooksMutationName = "createbooks";
+            string createMultipleBookMutation = @"mutation {
+                    createbooks(items: [{ title: ""Book #1"", publishers: { name: ""Publisher #1""}},
+                                        { title: ""Book #2"", publishers: { name: ""Publisher #2""}}]) {
+                        items{
+                            id
+                            title
+                        }
+                    }
+                }";
+
```

Comment:
another test to add would be to ensure that the whole request fails if even one of the inserts is forbidden:
```graphql
mutation {
                    createbooks(items: [{ title: ""Book #1"", publisher_id: 1234},
                                        { title: ""Book #2"", publishers: { name: ""Publisher #2""}}]) {
                        items{
                            id
                            title
                        }
                    }
                }";
```

Reply:
The test that you're suggesting would cover this test case as well. Added that.